toolkit/imageconfigs/read-only-root-efi.json

```diff
@@ -0,0 +1,101 @@
+{
+    "Disks": [
+        {
+            "PartitionTableType": "gpt",
+            "MaxSize": 5000,
+            "Artifacts": [
+                {
+                    "Name": "read-only",
+                    "Type": "vhdx"
+                },
+                {
+                    "Name": "read-only-raw",
+                    "Type": "raw"
+                }
+            ],
+            "Partitions": [
+                {
+                    "ID": "efi",
+                    "Flags": [
+                        "esp",
+                        "boot"
+                    ],
+                    "Start": 1,
+                    "End": 9,
+                    "FsType": "fat32"
+                },
+                {
+                    "ID": "boot",
+                    "Start": 9,
+                    "End": 400,
+                    "FsType": "ext4"
+                },
+                {
+                    "ID": "rootfs",
+                    "Start": 400,
+                    "End": 4000,
+                    "FsType": "ext4",
+                    "Flags": [
+                        "dmroot"
+                    ]
+                },
+                {
+                    "ID": "data",
+                    "Start": 4000,
+                    "End": 0,
+                    "FsType": "ext4"
+                }
+            ]
+        }
+    ],
+    "SystemConfigs": [
+        {
+            "Name": "ReadOnly",
+            "BootType": "efi",
+            "PartitionSettings": [
+                {
+                    "ID": "boot",
+                    "MountPoint": "/boot"
+                },
+                {
+                    "ID": "efi",
+                    "MountPoint": "/boot/efi",
+                    "MountOptions" : "umask=0077"
+                },
+                {
+                    "ID": "rootfs",
+                    "MountPoint": "/"
+                },
+                {
+                    "ID": "data",
+                    "MountPoint": "/data"
+                }
+            ],
+            "PackageLists": [
+                "packagelists/hyperv-packages.json",
+                "packagelists/core-packages-image.json",
+                "packagelists/read-only-root-packages.json"
+            ],
+            "KernelOptions": {
+                "default": "kernel"
+            },
+            "ReadOnlyVerityRoot": {
+                "Enable": true,
+                "TmpfsOverlays": [
+                    "/var"
+                ],
+                "TmpfsOverlayDebugEnabled": false
+            },
+            "KernelCommandLine": {
+                "ExtraCommandLine": "console=tty0 console=ttyS0=9800 rd.shell=0 rd.emergency=reboot"
+            },
+            "Hostname": "cbl-mariner",
+            "Users": [
+                {
+                    "Name": "root",
+                    "Password": "p@ssw0rd"
```

Comment:
Should this user have an explicitly set password?

Reply:
+1 I think we're trying to avoid the hard coded passwords here, but not sure if that's an option.

Reply:
Open to suggestions here. Unless you do something to setup a mount point or symlink (and I think shadow won't work with symlinks...) the user is also read-only.

I could switch to something SSH key related maybe?

---
In reply to: [552875082](https://github.com/microsoft/CBL-Mariner/pull/483#discussion_r552875082) [](ancestors = 552875082)

Reply:
I've switched to a hashed password which is a better example to set, but I've left the password in a comment along with a mention of the ssh key. I think this is a good compromise (Jon was ok with the password some time ago when I talked with him about it).

---
In reply to: [552939716](https://github.com/microsoft/CBL-Mariner/pull/483#discussion_r552939716) [](ancestors = 552939716,552875082)